f8a_report/report_helper.py

```diff
@@ -404,18 +405,16 @@ def normalize_worker_data(self, start_date, end_date, stack_data, worker, freque
                 'pypi': self.populate_key_count(stacks_list['pypi'])
             }
 
-            today = dt.today()
-            # Invoke this every Monday. In Python, Monday is 0 and Sunday is 6
-            # temp fix for retraining bug
-            if today.weekday() >= 0:
-                # Collate Data from Previous Month for Model Retraining
+            if retrain == 'T':
+                # Appends last week's data to 'collated-weekly'; returns 'BQ+collated' data
                 collated_data = self.collate_raw_data(unique_stacks_with_recurrence_count,
                                                       'weekly')
-                # Store ecosystem specific data to their respective Training Buckets
+                # Stores ecosystem specific manifest.json and re-trains models
                 self.store_training_data(collated_data)
+                return
```

Comment:
This return statement assumes that you will not do retraining and normal venus reporting together ever. It's either/or. Rather that that, you could have moved or broken this function down into two modules. Code before this if statement and after this if statement as two different functions. Then handle the if part accordingly.

Reply:
@miteshvp as discussed, working on breaking the functions.